pkg/plugin/plugin.go

```diff
@@ -95,7 +97,20 @@ func ParseError(err error) (errorType KMSErrorType) {
 	// ref. https://docs.aws.amazon.com/kms/latest/developerguide/requests-per-second.html
 	case kms.ErrCodeLimitExceededException:
 		return KMSErrorTypeThrottled
+
+	// AWS SDK Go for KMS does not "yet" define specific error code for a case where a customer specifies the deleted key
+	// "AccessDeniedException" error code may be returned when (1) CMK does not exist (not pending delete),
+	// or (2) corresponding IAM role is not allowed to access the key.
+	// Thus we only want to mark "AccessDeniedException" as user-induced for the case (1).
+	// e.g., "AccessDeniedException: The ciphertext refers to a customer master key that does not exist, does not exist in this region, or you are not allowed to access."
+	// KMS service may change the error message, so we do the string match.
+	case "AccessDeniedException":
+		if strings.Contains(ev.Message(), "customer master key that does not exist") ||
```

Comment:
Is `AccessDeniedException` not enough to detect a user induced error? String match is not a great way to handle this. I think `AccessDeniedException` is a good enough case to call an error user induced https://docs.aws.amazon.com/kms/latest/APIReference/CommonErrors.html

Reply:
The service may also return `AccessDeniedException` in case the encryption provider runs with invalid instance role (e.g., omitted `kms:Encrypt`), which is not user-induced.